tools/streamer.py

```diff
@@ -0,0 +1,8 @@
+import os
+from time import sleep 
+
+while True:
+  os.system("/home/pi/companion/scripts/start_video.sh $(cat /home/pi/vidformat.param)") 
```

Comment:
You can check if both files exist (`os.path.isfile`) before calling `os.system`. _better safe than sorry_
If one of the files does not exist, an error message should be printed and the script should return a error number different of 0.

Reply:
Both files do not need to exist :). Also, by the same logic we should check that every script exists before we run it. That seems like a bigger maintenance problem to me than just doing it the way we have done... this has never appeared as an issue in debugging.

Reply:
Should probably be using xargs here. But we should concentrate on moving to upstream it is very unclear how much of the code in this repository will remain as is in that effort. My instinct is telling me not much.

Reply:
if `/home/pi/companion/scripts/start_video.sh` does not exist, the loop will run at maximum speed putting the cpu in 100%.

Reply:
ok good point

Reply:
Added the os.path.isfile():

```
import os
from time import sleep

on=1
while True:
 
  while ((on<2):
    if (os.path.isfile("/home/pi/companion/start_video.sh")):
      os.system("/home/pi/companion/scripts/start_video.sh $(cat /home/pi/vidformat.param)")
    else:
      break 
    on += 1
    print "\nInitiated Camera"
    sleep(2)

  while (os.system("ls /dev/video* 2>/dev/null") != 0):
    if on==2:
      print "\nNo Video Device found"
      on = 1
    sleep(2)

  sleep (5)
```

Reply:
The main problem is that `start_video.sh` will start a daemonized screen. Making `os.system(...start_video.sh...` to finish and the python script to continue with the loop.
One suggestion to avoid this kind of problem, is to execute `start_video.sh` and getting the screen session [pid](https://en.wikipedia.org/wiki/Process_identifier), with that, it'll be possible to check if the screen is running, avoiding a second call of `start_video.sh`.

Besides that, a good practice while writing python scripts, it to check _how good_ they are with [pylint](https://www.pylint.org/) :woman_technologist:.

Reply:
The fix that i provided, avoids second call of start_video.sh until the camera is plugged out and plugged in again. Did you happen to run that? Meanwhile I will write a code with your method as well and run it.